tests/FixerFactoryTest.php

```diff
@@ -100,7 +100,7 @@ public function testRegisterBuiltInFixers(): void
             static function (string $className): bool {
                 $class = new \ReflectionClass($className);
 
-                return !$class->isAbstract() && $class->implementsInterface(FixerInterface::class) && str_starts_with($class->getNamespaceName(), 'PhpCsFixer\\Fixer\\');
+                return !$class->isAbstract() && !$class->isAnonymous() && $class->implementsInterface(FixerInterface::class) && str_starts_with($class->getNamespaceName(), 'PhpCsFixer\\Fixer\\');
```

Comment:
looking on code itself, I would get very confused.
Why this is needed? I see commitlog, but it doesn't indicate root cause.

Reply:
Let's see - [here](https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/compare/dx_remove_prophecy...6b7562617765726c6f73:PHP-CS-Fixer:dx_remove_prophecy_windows) are the changes  (`exit` instead of fail to actually show the output, thus build is passing), and [here](https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/actions/runs/7118779626/job/19382435174#step:6:460) the problem is.

Reply:
Do you know why it behaves differently on Windows?

John Doe, when reading the code, will still not understand why we had to put that new condition part.

btw, interesting to see that `exit` didn't fail the tests.

Reply:
> Do you know why it behaves differently on Windows?

https://github.com/php/php-src/issues/12895

Reply:
> John Doe, when reading the code, will still not understand why we had to put that new condition part.
IMHO we should at least address this part.

Reply:
I will forget it in a week, after we merge this I will create a draft PR to remove it and link it to that `php-src` issue.